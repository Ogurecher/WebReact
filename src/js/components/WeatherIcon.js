import React, { Component } from "react";

export default class WeatherIcon extends Component {
    constructor(props) {
        super(props);
    }
    
    render() {
        return (
            <img src={`http://openweathermap.org/img/wn/${this.props.weather.weather[0].icon}@2x.png`}/>
        );
    }
}