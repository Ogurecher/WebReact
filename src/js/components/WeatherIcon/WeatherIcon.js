import React, { Component } from 'react';
import { WEATHER_ICON_BASE_URL } from '../../constants/resources'

export default class WeatherIcon extends Component {

    render() {
        return (
            <img src={`${WEATHER_ICON_BASE_URL}${this.props.weather.weather[0].icon}@2x.png`}/>
        );
    }
}