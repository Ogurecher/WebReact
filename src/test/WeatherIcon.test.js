import React from 'react';
import renderer from 'react-test-renderer';

import WeatherIcon from '../js/components/WeatherIcon/WeatherIcon';

describe('WeatherIcon', function() {
    test('snapshot renders', function() {
        const weatherStub = {
            weather: [{
                icon: '50n'
            }]
        };

        const component = renderer.create(<WeatherIcon weather={weatherStub} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});