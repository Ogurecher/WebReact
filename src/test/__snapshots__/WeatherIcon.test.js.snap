// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIcon snapshot renders 1`] = `
<img
  src="http://openweathermap.org/img/wn/50n@2x.png"
/>
`;
